gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala

```diff
@@ -159,74 +158,48 @@ abstract class HashAggregateExecBaseTransformer(
   // Members declared in org.apache.spark.sql.execution.AliasAwareOutputPartitioning
   override protected def outputExpressions: Seq[NamedExpression] = resultExpressions
 
+  // Check if Pre-Projection is needed before the Aggregation.
   protected def needsPreProjection: Boolean = {
-    var needsProjection = false
-    breakable {
-      for (expr <- groupingExpressions) {
-        if (!expr.isInstanceOf[Attribute]) {
-          needsProjection = true
-          break
-        }
-      }
-    }
-    breakable {
-      for (expr <- aggregateExpressions) {
-        if (
-          expr.filter.isDefined && !expr.filter.get.isInstanceOf[Attribute] &&
-          !expr.filter.get.isInstanceOf[Literal]
-        ) {
-          needsProjection = true
-          break
+    groupingExpressions.exists {
+      case _: Attribute => false
+      case _ => true
+    } || aggregateExpressions.exists {
+      expr =>
+        expr.filter match {
+          case None =>
+          case Some(_: Attribute) =>
+          case Some(_: Literal) =>
```

Comment:
Don't we need return value for these 3 branches?

Reply:
The original code, when encountering `expr.filter` defined and it is not `Attribute` and `Literal`, would set needsProjection to true and break to return directly. This corresponds to the modified code's `case _ => return true`. 
In other cases, it will not immediately break, but continue the next validation based on `expr.mode`. If these three cases explicitly return false, the subsequent validation will be skipped. If there is no return value, it will proceed to the validation based on `expr.mode` below.

Reply:
Seems the CI fails. Could you take a look?